File: services/customers-service/src/main/java/br/com/deltaglobalbank/customers/domain/customer/valueobjects/BirthDate.java
package br.com.deltaglobalbank.customers.domain.customer.valueobjects;

import java.time.LocalDate;

public record BirthDate(LocalDate value) {
    public BirthDate {
        LocalDate minAgeDate = LocalDate.now().minusYears(18);
        if (value.isAfter(minAgeDate)) {
            throw new IllegalArgumentException("customer_underage");
        }
    }
}

File: services/customers-service/src/main/kotlin/br/com/deltaglobalbank/customers/domain/customer/valueobjects/BirthDate.kt
package br.com.deltaglobalbank.customers.domain.customer.valueobjects

import java.time.LocalDate

@JvmInline
value class BirthDate(val value: LocalDate) {
    init {
        val minAgeDate = LocalDate.now().minusYears(18)
        require(!value.isAfter(minAgeDate)) { "customer_underage" }
    }
}